services/auth/source/oauth2/providers_base.go

```diff
@@ -24,6 +24,17 @@ func (b *BaseProvider) Image() string {
 	return "/assets/img/auth/" + b.name + ".png"
 }
```

Comment:
Is this unused now?

Reply:
It is unused, but I am not sure if it should be removed on a design level.

Reply:
And `NewImagedProvider` is also not used after https://github.com/go-gitea/gitea/pull/25030/commits/ed03bf51f698db913f2889bf566390e87b0c2f32

Reply:
I see no point in keeping unused methods around.

Reply:
Agree to remove it.

Reply:
After reading the code, I think this "Image()" can't be removed. 

And I guess some PRs broke the "auth icon" system


Users could set IconURL when editing an OAuth2 auth source, that image should be used, but not the svg icon.

Reply:
Added back `Image()` and removed `SvgName()`, but used `svg` instead of `png` as default icon images, please review again, thanks.